### PR TITLE
Tidying, compatability and CLI usability

### PR DESCRIPTION
**Pull request by AJFaraday**
- Ruby 2.2.1 compatibility
- Some refactors and reformatting
- Trailing slash no longer required in arguments
- STDOUT sync enabled to allow CLI piping
- Block only fires for successfully retrieved URLS now fire actions
- Tidied README and documented pipeable functionality

### CODE REVIEW

**Comment by robmiller** [comment]
Yikes, this is a blast from the past! Thanks. I'll drop a couple of comments inline.

File: lib/redback.rb [hunk 1]
```diff
@@ -31,6 +30,13 @@ def initialize(url, &each_site)
     spider
   end
 
+  def fix_url_formatting(url)
+    url = url.strip
+    url = "http://#{url}" unless url[/^http:\/\//] || url[/^https:\/\//]
+    url = "#{url}/" unless url.strip[-1] == "/"
```
Comment (robmiller) [inline lib/redback.rb]: Unnecessary strip?

File: lib/redback.rb [hunk 2]
```diff
@@ -44,34 +50,33 @@ def crawl_page(url, limit = 10)
 
     begin
       uri = URI.parse(URI.encode(url.to_s.strip))
-    rescue
+    rescue => er
+      puts "Could not parse URL: #{url}"
+      puts er.message
+      puts er.backtrace.join("\n")
```
Comment (robmiller) [inline lib/redback.rb]: `$stderr.puts` is better here IMO

File: lib/redback.rb (hunk 1 above)
Comment (AJFaraday) [inline lib/redback.rb]: When used within ruby a trailing or leading space was confusing the URI parser.

Comment (robmiller) [inline lib/redback.rb]: But the URL is already stripped (line 34), so it can't have leading or trailing whitespace at this point.

Comment (AJFaraday) [inline lib/redback.rb]: Correct, I missed that, sorry.

**Comment by AJFaraday** [comment]
Fix the issues mentioned in your comments.

**Comment by robmiller** [comment]
Thanks!

**Pull request merged**